File: CSharpStuff/Nathandelane.Math.Processor/Nathandelane.Math.Processor.Tokens/IToken.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Nathandelane.Math.Processor.Tokens
{
    public interface IToken
    {
        #region Propeties

        TokenType Type { get; set; }
        TokenPrecedence Precedence { get; set; }
        string Value { get; set; }

        #endregion
    }
}

File: JavaStuff/PersonalCalculator/personalcalculator/src/com/nathandelane/math/evaluation/tokens/IToken.java
package com.nathandelane.math.evaluation.tokens;

public interface IToken {
	
	public boolean matches(String pattern) throws NotImplementedException;
	
	public String firstMatch() throws NotImplementedException;
	
	public TokenType getType();
	
	public TokenPrecedence getPrecedence();
	
	public void setPrecedence(TokenPrecedence precedence);
	
	public String getValue();
	
	public String toString();
}
